cmd/local-gadget/containers/containers.go

```diff
@@ -127,65 +82,30 @@ func NewListContainersCmd() *cobra.Command {
 		"Don't truncate container ID",
 	)
 
-	utils.AddCommonFlags(cmd, commonFlags)
+	utils.AddCommonFlags(cmd, &commonFlags)
 
 	return cmd
 }
 
-func (p *ContainerParser) SortContainers(containers []*containercollection.Container) {
-	sort.Slice(containers, func(i, j int) bool {
-		si, sj := (containers)[i], (containers)[j]
-		switch {
-		case si.Runtime != sj.Runtime:
-			return si.Runtime < sj.Runtime
-		case si.Name != sj.Name:
-			return si.Name < sj.Name
-		default:
-			return si.ID < sj.ID
-		}
-	})
-}
+func newContainerParser(
+	outputConfig *commonutils.OutputConfig, flags *ContainerFlags,
+) (
+	*commonutils.GadgetParser[containercollection.Container], error,
+) {
+	containerColumns := containercollection.GetColumns()
 
-func (p *ContainerParser) TransformToColumns(c *containercollection.Container) string {
-	var sb strings.Builder
-
-	for _, col := range p.OutputConfig.CustomColumns {
-		switch col {
-		case "runtime":
-			sb.WriteString(fmt.Sprintf("%s", c.Runtime))
-		case "id":
-			if p.containerFlags.noTrunc {
-				sb.WriteString(fmt.Sprintf("%s", c.ID))
-			} else {
-				sb.WriteString(fmt.Sprintf("%.13s", c.ID))
-			}
-		case "name":
-			sb.WriteString(fmt.Sprintf("%s", c.Name))
-		case "pid":
-			sb.WriteString(fmt.Sprintf("%d", c.Pid))
-		case "mntns":
-			sb.WriteString(fmt.Sprintf("%d", c.Mntns))
-		case "netns":
-			sb.WriteString(fmt.Sprintf("%d", c.Netns))
-		case "namespace":
-			sb.WriteString(fmt.Sprintf("%s", c.Namespace))
-		case "podname":
-			sb.WriteString(fmt.Sprintf("%s", c.Podname))
-		case "poduid":
-			sb.WriteString(fmt.Sprintf("%s", c.PodUID))
-		case "cgrouppath":
-			sb.WriteString(fmt.Sprintf("%s", c.CgroupPath))
-		case "cgroupid":
-			sb.WriteString(fmt.Sprintf("%d", c.CgroupID))
-		case "cgroupv1":
-			sb.WriteString(fmt.Sprintf("%s", c.CgroupV1))
-		case "cgroupv2":
-			sb.WriteString(fmt.Sprintf("%s", c.CgroupV2))
-		default:
-			continue
-		}
-		sb.WriteRune('\t')
+	idCol, ok := containerColumns.GetColumn("id")
+	if !ok {
+		return nil, fmt.Errorf(`"id" column doesn't exist`)
+	}
+
+	// By default, the "id" column is configured with width=13 and no fixed. In
+	// this way, the auto-scale will do its magic when --no-trunc is not set.
+	// Instead, when it is set, we must ensure auto-scale doesn't truncate it.
+	if flags.noTrunc {
+		idCol.Width = 64
+		idCol.FixedWidth = true
```

Comment:
This isn't needed anymore now that the table is adjusted by content, right? But maybe use the noTrunc as parameter to the AdjustByContent() to negate the `force` parameter, so that when the terminal is too small, it would output everything nevertheless.